tests/kernel/mem_protect/userspace/src/mem_domain.c

```diff
@@ -0,0 +1,333 @@
+/* 
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <ztest.h>
+#include <kernel_structs.h>
+#include <string.h>
+#include <stdlib.h>
+
+#define MEM_DOMAIN_STACK_SIZE 1024
+K_THREAD_STACK_DEFINE(mem_domain_1_stack, MEM_DOMAIN_STACK_SIZE);
+K_THREAD_STACK_DEFINE(mem_domain_2_stack, MEM_DOMAIN_STACK_SIZE);
+K_THREAD_STACK_DEFINE(mem_domain_5_stack, MEM_DOMAIN_STACK_SIZE);
+__kernel struct k_thread mem_domain_1_tid, mem_domain_2_tid, mem_domain_5_tid;
+bool valid_fault;
+
+/* variables */
+#ifdef CONFIG_ARM
+u8_t __aligned(32) mem_domain_buf[32]; /* for mem domain */
+
+u8_t __aligned(32) mem_domain_tc3_part1[32];
+u8_t __aligned(32) mem_domain_tc3_part2[32];
+u8_t __aligned(32) mem_domain_tc3_part3[32];
+u8_t __aligned(32) mem_domain_tc3_part4[32];
+u8_t __aligned(32) mem_domain_tc3_part5[32];
+u8_t __aligned(32) mem_domain_tc3_part6[32];
+u8_t __aligned(32) mem_domain_tc3_part7[32];
+u8_t __aligned(32) mem_domain_tc3_part8[32];
+#else
+
+u8_t __aligned(4096) mem_domain_buf[4096]; /* for mem domain */
+
+u8_t __aligned(4096) mem_domain_tc3_part1[4096];
+u8_t __aligned(4096) mem_domain_tc3_part2[4096];
+u8_t __aligned(4096) mem_domain_tc3_part3[4096];
+u8_t __aligned(4096) mem_domain_tc3_part4[4096];
+u8_t __aligned(4096) mem_domain_tc3_part5[4096];
+u8_t __aligned(4096) mem_domain_tc3_part6[4096];
+u8_t __aligned(4096) mem_domain_tc3_part7[4096];
+u8_t __aligned(4096) mem_domain_tc3_part8[4096];
+#endif
+
+K_MEM_PARTITION_DEFINE(mem_domain_memory_partition, mem_domain_buf, sizeof(mem_domain_buf),
+                       K_MEM_PARTITION_P_RW_U_RW);
+
+struct k_mem_partition *mem_domain_memory_partition_array[] = {
+	&mem_domain_memory_partition
+};
+
+struct k_mem_domain mem_domain_mem_domain;
+
+/****************************************************************************/
+/* Common init functions */
+static inline void mem_domain_init(void)
+{
+	k_mem_domain_init(&mem_domain_mem_domain,
+			  1,
+			  mem_domain_memory_partition_array);
+}
+
+static inline void set_valid_fault_value(int test_case_number)
+{
+	switch(test_case_number){
+	case 1:
+		valid_fault = false;
+		break;
+	case 2:
+		valid_fault = true;
+		break;
+	case 3:
+		valid_fault = false;
+		break;
+	case 4:
+		valid_fault = true;
+		break;
+	case 5:
+		break;
+	case 6:
+		valid_fault = true;
+		break;		
+		}
+}
```

Comment:
This seems very fragile; keeping this in sync with the actual test cases will be a headache.  Why is this better than just directly setting valid_fault in each test case where we can see whether a fault is legitimate or not.

Reply:
makes sense. I will change it